src/entity.ts

```diff
@@ -376,11 +435,49 @@ export namespace entity {
     return value instanceof entity.Key;
   }
 
+  /**
+   * Convert a protobuf `integerValue`.
+   *
+   * @private
+   * @param {object} value The `integerValue` to convert.
+   */
+  function decodeIntegerValue(value: ValueProto) {
+    const num = Number(value.integerValue);
+    if (!Number.isSafeInteger(num)) {
+      throw new Error(
+        'We attempted to return all of the numeric values, but ' +
+          (value.propertyName ? value.propertyName + ' ' : '') +
+          'value ' +
+          value.integerValue +
+          " is out of bounds of 'Number.MAX_SAFE_INTEGER'.\n" +
+          "Please consider passing 'options.wrapNumbersOptions=true' or\n" +
```

Comment:
I like that you went with a dual-purpose option, where it can be a boolean, for simple mode, or an object for an advanced use case. I would rather have `wrapNumbers` be the name, however, so that `wrapNumbers: true` is possible, compared to `wrapNumbersOptions: true`. When the object is used, `wrapNumbers = {integerTypeCastFunction: () => {}}` still seems fine.

Reply:
Renamed